src/store/modules/agent.js

```diff
@@ -223,9 +237,66 @@ export default ({
     },
     setIsDefaultViewOnAgent (state, value) {
       state.isDefaultViewOnAgent = value
+    },
+    updateAuthenticationToken (state, token) {
+      state.authentication_token = token
     }
   },
   actions: {
+    login ({ commit, state }, credentials) {
+      return axios.post('/auth/login',
+        {
+          userName: credentials.username,
+          password: credentials.password
+        })
+        .then(function (response) {
+          commit('updateAuthenticationToken', response.data.auth_token)
+          axios.defaults.headers.common.Authorization = 'Bearer ' + state.authentication_token
+          return true
+        })
+        .catch(function () {
+          return false
+        })
+    },
+    loadResources ({ state, commit }) {
```

Comment:
we can move the resources actions to a file src/store/modules/referent.js instead in the agent.js